base/env.jl

```diff
@@ -100,7 +100,7 @@ if Sys.iswindows()
         if m === nothing
             error("malformed environment entry: $env")
         end
-        return (Pair{String,String}(m.captures[1], m.captures[2]), (pos+(len+1)*2, blk))
+        return (Pair{String,String}(uppercase(m.captures[1]), m.captures[2]), (pos+(len+1)*2, blk))
```

Comment:
Should be `ASCII.uppercase` (do we have such a thing?)

Reply:
I just checked, and Windows is also case-insensitive for Unicode characters, e.g. [U+00ee](http://www.fileformat.info/info/unicode/char/00EE/index.htm) vs. [U+00ce](http://www.fileformat.info/info/unicode/char/00CE/index.htm), so arguably we should uppercase those as well.

Reply:
Actually, I take that back, I have no idea what table is used by windows (I'm guessing that it isn't well-defined or stable or consistent)

libuv uses `LCMapStringW(LOCALE_INVARIANT, LCMAP_UPPERCASE)`
MS doesn't really say how the comparison is done: https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/wdm/nf-wdm-rtlcompareunicodestring

Reply:
Okay, updated to use `LCMapStringW`.